functional/pipeline.py

```diff
@@ -179,13 +183,10 @@ def _transform(self, *transforms):
         :param transform: transform to apply or list of transforms to apply
         :return: transformed sequence
         """
-        sequence = None
-        for transform in transforms:
-            if sequence:
-                sequence = Sequence(sequence, transform=transform)
-            else:
-                sequence = Sequence(self, transform=transform)
-        return sequence
+        return reduce(
```

Comment:
I am not sure how that would be accomplished. In general, I prefer the explicit for loop for readability so without a compelling reason won't change it to use `reduce`